net/validators_test.go

```diff
@@ -0,0 +1,148 @@
+package net_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/libp2p/go-libp2p-pubsub"
+	"github.com/libp2p/go-libp2p-pubsub/pb"
+	mocknet "github.com/libp2p/go-libp2p/p2p/net/mock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/go-filecoin/address"
+	"github.com/filecoin-project/go-filecoin/chain"
+	"github.com/filecoin-project/go-filecoin/consensus"
+	"github.com/filecoin-project/go-filecoin/net"
+	th "github.com/filecoin-project/go-filecoin/testhelpers"
+	tf "github.com/filecoin-project/go-filecoin/testhelpers/testflags"
+	"github.com/filecoin-project/go-filecoin/types"
+)
+
+func TestBlockTopicValidator(t *testing.T) {
+	tf.UnitTest(t)
+
+	ctx := context.Background()
+	mbv := th.NewMockBlockValidator()
+	tv := net.NewBlockTopicValidator(mbv, nil)
+	builder := chain.NewBuilder(t, address.Undef)
+	pid1 := th.RequireIntPeerID(t, 1)
+
+	goodBlk := builder.BuildOnBlock(nil, func(b *chain.BlockBuilder) {})
+	badBlk := builder.BuildOnBlock(nil, func(b *chain.BlockBuilder) {
+		b.IncHeight(1)
+	})
+
+	mbv.StubSyntaxValidationForBlock(badBlk, fmt.Errorf("invalid block"))
+
+	validator := tv.Validator()
+
+	assert.Equal(t, net.BlockTopic, tv.Topic())
+	assert.True(t, validator(ctx, pid1, blkToPubSub(goodBlk)))
+	assert.False(t, validator(ctx, pid1, blkToPubSub(badBlk)))
+	assert.False(t, validator(ctx, pid1, nonBlkPubSubMsg()))
+}
+
+func TestBlockPubSubValidation(t *testing.T) {
+	tf.IntegrationTest(t)
+	ctx := context.Background()
+
+	// setup a mock network and generate a host
+	mn := mocknet.New(ctx)
+	host1, err := mn.GenPeer()
+	require.NoError(t, err)
+
+	// create a fake clock to trigger block validation failures
+	now := time.Unix(1234567890, 0)
+	mclock := th.NewFakeSystemClock(now)
+	// block time will be 1 second
+	blocktime := time.Second * 1
+
+	// setup a block validator and a topic validator
+	bv := consensus.NewDefaultBlockValidator(blocktime, mclock)
+	btv := net.NewBlockTopicValidator(bv)
+
+	// setup a floodsub instance on the host and register the topic validator
+	fsub1, err := pubsub.NewFloodSub(ctx, host1, pubsub.WithMessageSigning(false))
+	require.NoError(t, err)
+	err = fsub1.RegisterTopicValidator(btv.Topic(), btv.Validator(), btv.Opts()...)
+	require.NoError(t, err)
```

Comment:
Thanks for this test, it's a good start.

I was hoping to exercise the logic in the node that does this hookup (e.g. via `GenNode`), rather than replicate it here. Right now this test is checking that libp2p does what it promises, which is not without value, but also not testing much of our code. 

Let's leave this test here, but could you add to the comments that the goal is to instantiate a node and check that it receives but rejects the invalid block header. That will require some more precise control over the node than is currently easy (e.g. injecting the fake clock), but I'm working on it!